lib/command/run.rb

```diff
@@ -48,79 +74,241 @@ class Run < Base
+
+        # Create runner workload
+        cp.apply_hash("kind" => "workload", "name" => runner_workload, "spec" => spec)
+      end
+    end
+
+    def update_runner_workload
+      step("Updating runner workload '#{runner_workload}' based on '#{original_workload}'") do
+        spec, container_spec = base_workload_specs(runner_workload)
+
+        # Override image if specified
+        image = config.options[:image]
+        image = latest_image if image == "latest"
+        container_spec["image"] = "/org/#{config.org}/image/#{image}" if image
+
+        # Container overrides
+        container_spec["cpu"] = config.options[:cpu] if config.options[:cpu]
+        container_spec["memory"] = config.options[:memory] if config.options[:memory]
+
+        # Update runner workload
+        cp.apply_hash("kind" => "workload", "name" => runner_workload, "spec" => spec)
+      end
+    end
+
+    def wait_for_runner_workload
+      step("Waiting for runner workload '#{runner_workload}' to be created", retry_on_failure: true) do
+        cp.fetch_workload(runner_workload)
       end
+    end
+
+    def start_job
+      job_start_yaml = build_job_start_yaml
+
+      step("Starting job for runner workload '#{runner_workload}'", retry_on_failure: true) do
+        result = cp.start_cron_workload(runner_workload, job_start_yaml, location: location)
+        @job = result&.dig("items", 0, "id")
 
-      # Create workload clone
-      cp.apply_hash("kind" => "workload", "name" => workload_clone, "spec" => spec)
+        job || false
+      end
     end
 
-    def runner_script # rubocop:disable Metrics/MethodLength
-      script = Scripts.helpers_cleanup
+    def wait_for_replica_for_job
+      step("Waiting for replica to start, which runs job '#{job}'", retry_on_failure: true) do
+        result = cp.fetch_workload_replicas(runner_workload, location: location)
+        @replica = result["items"].find { |item| item.include?(job) }
 
-      if config.options["use_local_token"]
-        script += <<~SHELL
-          CPLN_TOKEN=$CONTROLPLANE_TOKEN
-          unset CONTROLPLANE_TOKEN
-        SHELL
+        replica || false
       end
+    end
+
+    def run_interactive
+      progress.puts("Connecting to replica '#{replica}'...\n\n")
+      cp.workload_exec(runner_workload, replica, location: location, container: container, command: command)
+    end
+
+    def run_non_interactive # rubocop:disable Metrics/MethodLength
+      if detached
+        print_detached_commands
+        exit(ExitCode::SUCCESS)
+      end
+
+      logs_pid = Process.fork do
+        # Catch Ctrl+C in the forked process
+        trap("SIGINT") do
+          exit(ExitCode::SUCCESS)
+        end
+
+        Cpl::Cli.start(["logs", *app_workload_replica_args])
+      end
+      Process.detach(logs_pid)
+
+      # We need to wait a bit for the logs to appear,
+      # otherwise it may exit without showing them
+      Kernel.sleep(30)
```

Comment:
Perhaps there's a better way to ensure that all logs have been shown instead of waiting an arbitrary amount of seconds?